### ISSUE TITLE
Keep (at least a few) old versions in the apt repository

### ISSUE THREAD

**Issue by johnmarkwayve** [bazelbuild/bazel#6824]
### Description of the problem / feature request:

Please keep at least a few older releases available in the apt repository.

### Feature requests: what underlying problem are you trying to solve with this feature?

AFAICT, bazel only maintains the latest release in the apt repository. This means that if bazel updates with a breaking change and some users in your company have it installed via apt, then they will be irreversibly updated to the latest bazel version on an `apt upgrade`. This gives you zero time to roll out any fixes you need to update against breaking changes before they break peoples builds.

If bazel kept a few (maybe four or so) older releases in the apt repository, then you could simply tell affected users to downgrade their installation to one that works while a fix for your build files are worked on, then let them know they can safely upgrade when the fixes are out. This would make working with bazel so much smother across the organisation.

I like the fact that people installing from apt will, by default, get the latest and greatest as it comes out. But it can be very frustrating when the latest and greatest gives me zero time to roll out updates to our build configuration with no easy way for users to temporarily downgrade to a working version.

**Comment by kamahen** [comment]
It's easy to make a mistake and upgrade Bazel even when I know the latest version is broken, because the typical way of upgrading is
```bash
$ sudo apt-get update && sudo apt-get --with-new-pkgs upgrade 
```
and it's easy to not notice that `bazel` is in the list of upgraded packages.

And there's only one version in the PPA:
```
$ apt list -a bazel
Listing... Done
bazel/unknown,now 0.25.0 amd64 [installed]
```

**Comment by philwo** [comment]
I'm working on fixing this right now. We will provide all older Bazel versions in our apt-get repository in addition to the latest one and probably also make it possible to install versions in parallel (e.g. Bazel 0.29.1 can be installed side-by-side as `/usr/bin/bazel-0.29.1`).

Comments on the latter idea are welcome, whether that would be useful for you.

**Comment by johnmarkwayve** [comment]
For the moment this is no longer an issue for my team, as we're using [bazelisk](github.com/bazelbuild/bazelisk) to manage versions now.

However I still think this is worth fixing (and the side-by-side install support also makes sense to me)

**Comment by philwo** [comment]
@meteorcloudy fixed this in https://github.com/bazelbuild/bazel/commit/31627057454717d2216da0791969207569a1f423. :) Our apt repository now contains all stable version released since 1.0.0 and will continue to do so in the future
